lib/ddtrace/contrib/action_cable/instrumentation.rb

```diff
@@ -0,0 +1,34 @@
+module Datadog
+  module Contrib
+    module ActionCable
+      module Instrumentation
+        # Instruments ActionCable::Connection to ensure the top-level
+        # rack request has the correct resource for this WebSockets connection.
+        #
+        # Some of our common assumptions don't hold under WebSockets, one example
+        # being the http status code, which is -1 for this request.
+        module ActionCableConnection
+          def on_open
+            Datadog.tracer.trace('action_cable.on_open') do |span|
+              begin
+                span.resource = "#{self.class}#on_open"
+                span.span_type = Datadog::Ext::AppTypes::WEB
+
+                span.set_tag('action_cable.action', 'on_open')
+                span.set_tag('action_cable.connection', self.class.to_s)
+
+                # Set the resource name of the Rack request span
+                rack_request_span = env[Datadog::Contrib::Rack::TraceMiddleware::RACK_REQUEST_SPAN]
+                rack_request_span.resource = span.resource if rack_request_span
```

Comment:
Why are we modifying the Rack request span?

Reply:
I updated the comment block on this module to explain why:
```
When a new WebSocket is open, we receive a Rack request resource name "GET -1".
This module overrides the current Rack resource name to provide a meaningful name.
```

It changes it from this:
![Screen Shot 2019-10-08 at 6 04 56 PM](https://user-images.githubusercontent.com/583503/66414594-06ff4800-e9fa-11e9-8195-b68207bde6cd.png)

to this:
![Screen Shot 2019-10-08 at 6 33 26 PM](https://user-images.githubusercontent.com/583503/66414654-239b8000-e9fa-11e9-8b81-b483e31a7903.png)